lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -262,10 +262,11 @@ SPIRVToLLVMDbgTran::transTypeArrayOpenCL(const SPIRVExtInst *DebugInst) {
                                                        UpperBound, nullptr));
       continue;
     }
-    if (auto *ExprUB = getDbgInst<SPIRVDebug::Expression>(Ops[I])) {
+    if (const SPIRVExtInst * ExprUB, *ExprLB;
+        (ExprUB = getDbgInst<SPIRVDebug::Expression>(Ops[I])) &&
+        (ExprLB =
+             getDbgInst<SPIRVDebug::Expression>(Ops[Ops.size() / 2 + I]))) {
```

Comment:
While this check seems correct - it rings a bell, that something might be missing in SPIR-V generation part and we are losing debug information. May I ask you to add a test for this patch? It will also help understanding what might be missing.